CRM/Logging/ReportDetail.php

```diff
@@ -488,4 +510,83 @@ private function convertForeignKeyValuesToLabels(string $fkClassName, string $fi
     return (string) $keyval;
   }
 
+  /**
+   * Given a custom field with a numeric key value used as a foreign key, 
+   * return the "label" for the key value. 
+   * If the label doesn't exist, just echo back
+   * what we were given.
+   *
+   * @param int $keyval
+   *   This is the numeric key value id used to lookup its corresponding label.
+   * @param array $cfArray
+   *   This is the custom field as an array
+   * @return string
+   */
+  private function convertCustomFieldForeignKeyValuesToLabels(int $keyval, array $cfArray): string {
```

Comment:
There's a function `civicrm_api3('CustomValue', 'getdisplayvalue', ...` that handles getting the display value for a custom field. Would it be better to reuse that here?

Reply:
Yes, that makes more sense. I just made a new commit using the civicrm_api3 call. Thanks!

Reply:
Thanks, will take a look.